tracer/src/Datadog.Trace/Configuration/TracerSettings.cs

```diff
@@ -141,13 +141,23 @@ internal TracerSettings(IConfigurationSource? source, IConfigurationTelemetry te
             GlobalTagsInternal = config
                          // backwards compatibility for names used in the past
                         .WithKeys(ConfigurationKeys.GlobalTags, "DD_TRACE_GLOBAL_TAGS")
-                        .AsDictionary()
-                       // Filter out tags with empty keys or empty values, and trim whitespace
-                       ?.Where(kvp => !string.IsNullOrWhiteSpace(kvp.Key) && !string.IsNullOrWhiteSpace(kvp.Value))
-                        .ToDictionary(kvp => kvp.Key.Trim(), kvp => kvp.Value.Trim())
+                        .AsDictionary(&GlobalTagsSelector)
                       // default value (empty)
                       ?? (IDictionary<string, string>)new ConcurrentDictionary<string, string>();
 
+            // Filter out tags with empty keys or empty values, and trim whitespace
+            static bool GlobalTagsSelector(ref string key, ref string value)
+            {
+                if (string.IsNullOrWhiteSpace(key) || string.IsNullOrWhiteSpace(value))
+                {
+                    return false;
+                }
+
+                key = key.Trim();
+                value = value.Trim();
+                return true;
+            }
```

Comment:
Is this the only place we use function pointer? Is it really worth the extra complexity, given this is a one-off call? 🤔 Hint: No IMO 😛